### DISCUSSION TITLE
What exactly happens when you feed the model with long documents ?

### DISCUSSION THREAD

**Discussion by reouvenzana** [MaartenGr/BERTopic#822] [Q&A] [answered]
Hello!

I'm not familiar with how BERT encodes sentences and documents. I used BERTopic with documents of various lengths, some of wich being really long (>5000 words and more). I was pretty happy with the topics I got. Then I tried to split each document by sentences and passing them to the model, but the computational cost seems too high (the model has been running for ten hours as I'm writing this). So, could you please explain what I'm missing by feeding the model long documents? Are only the first _n_ tokens being considered? 

I should point out that I'm thinking about splitting the documents in paragraphs rather than sentences, but it will take time.

**Comment by MaartenGr** [comment, marked as answer]
It depends on the embedding model that you are using in BERTopic. If you are using the default model, which is a [sentence-transformer model](https://www.sbert.net/docs/pretrained_models.html), then the document gets truncated after a fixed number of tokens. This means that it will ignore a part of the document. 

When documents are long but also contain multiple topics, then it is advised to split them up into either paragraphs or sentences. If the computational cost is too high, then there are a number of tricks you can do. First, you can train the model on a subset of the data and predict the topics for every document outside of that subset. This allows you to circumvent having to train on the entire dataset. Second, you can use [online topic modeling](https://maartengr.github.io/BERTopic/getting_started/online/online.html) to train a model on batches of data instead of on all the data at once. Third, you can use different sub-models that might be more optimized for your use case. For example, if you have a GPU, then it might be worthwhile to use [cuML](https://maartengr.github.io/BERTopic/getting_started/tips_and_tricks/tips_and_tricks.html#gpu-acceleration) instead of the default models.

**Reply by reouvenzana** [reply]
Thank you for taking the time to answer. Great work by the way!
